### ISSUE TITLE
Please consider disallow returning values from a proptest 

### ISSUE THREAD

**Issue by shepmaster** [AltSysrq/proptest#154]
This will happily spin for a while and never fail:

```rust
use proptest::prelude::*;

proptest! {
    #[test]
    fn test_dummy(a: i32) {
        false
    }
}
```

This is the reduced version of my tests which I ported from quickcheck. I’ve been slowly introducing bugs because I forgot to change the returned boolean into an assert. This is something that a “normal” Rust function would have prevented because my function return type doesn’t match the returned value.

**Comment by shepmaster** [comment]
Looking at the expanded code:

```rust
            |::proptest::sugar::NamedArguments(_, a)| {
                {
                    true
                };
                Ok(())
            },
```

Perhaps you can change it to 

```rust
            |::proptest::sugar::NamedArguments(_, a)| {
                let _: () = {
//              ^^^^^^^^^^^
                    true
                };
                Ok(())
            },
```

**Comment by AltSysrq** [comment]
Duplicate of https://github.com/AltSysrq/proptest/issues/142

tl;dr Your code isn't intended to work but it does because the macro fails to ensure the body evaluates to the correct type (which I see you just discovered yourself). I'm planning on fixing that in the next release with breaking changes.

**Comment by shepmaster** [comment]
> I'm planning on fixing that in the next release with breaking changes.

Is there something preventing the proposed change from being deployed in a patch release immediately? This is a total killer and I’m frankly really disappointed that I’ve been trusting tests that aren’t testing anything and releasing bad code because of if. I’m quite worried that other people have the same problem.

**Comment by Andlon** [comment]
Just wanted to add that I had the exact same issue. I ported some tests from `quickcheck`, and it turned out that some of them were not actually testing anything at all, which as @shepmaster pointed out is very unfortunate. Looking forward to a new release with the fix!
